### PR TITLE
Fix #3238 -- MathOptInterface -> MOI

### LINKED ISSUE #3238: Depend on `MathOptInterface` directly instead of referencing the internal variable of `JuMP`
Currently, `MathOptInterface` is loaded through the `JuMP` module scope:

https://github.com/JuliaReach/LazySets.jl/blob/b11803321cd10ce904198d8abcf19759f9257f1d/src/Initialization/init_JuMP.jl#L1

The latest release of `JuMP` has renamed `MathOptInterface` internally to MOI. Thus, `LazySets` now fails to load.

I suggested in https://github.com/jump-dev/JuMP.jl/issues/3233 to keep an alias around until a breaking release is tagged. Nonetheless, it would be good to make the changes here now to depend on `MathOptInterface` directly instead of "stealing" it from `JuMP`.

### PR DESCRIPTION
**Pull request by lassepe**

### CODE REVIEW

**Comment by lassepe** [comment]
As per the discussion in https://github.com/jump-dev/JuMP.jl/issues/3233, accessing MathOptInterface as `JuMP.MOI` variable is encouraged and also works on older JuMP version.

**Review by schillic — APPROVED** [review]
Thanks for the proposal! Looks good to me.

**Comment by schillic** [comment]
The new release is out.
CC @BenCurran98

File: src/Initialization/init_JuMP.jl [hunk 1]
```diff
@@ -1,34 +1,34 @@
-using JuMP.MathOptInterface: AbstractOptimizer, OptimizerWithAttributes
+using JuMP.MOI: AbstractOptimizer, OptimizerWithAttributes
 using JuMP: Model, @variable, @objective, @constraint, optimize!,
             termination_status, objective_value, value, primal_status
 
 # solver status
 function is_lp_optimal(status)
-    return status == JuMP.MathOptInterface.OPTIMAL
+    return status == JuMP.MOI.OPTIMAL
```
Comment (odow) [inline src/Initialization/init_JuMP.jl:7]: This can be `JuMP.OPTIMAL`, and so forth

File: src/Initialization/init_JuMP.jl [hunk 2]
```diff
@@ -1,34 +1,34 @@
-using JuMP.MathOptInterface: AbstractOptimizer, OptimizerWithAttributes
+using JuMP.MOI: AbstractOptimizer, OptimizerWithAttributes
```
Comment (odow) [inline src/Initialization/init_JuMP.jl:1]: If you need these, you should probably instead declare a `Pkg` dependency on `MathOptInterface` explicitly? It's weird to import structs/methods from a dependency of a package.

Comment (mforets) [reply src/Initialization/init_JuMP.jl:1]: thanks, done in https://github.com/JuliaReach/LazySets.jl/pull/3245#pullrequestreview-1346354090

**Pull request merged**